vulnerabilities/cve_2013_4164_spec.rb

```diff
@@ -0,0 +1,19 @@
+require File.expand_path('../../spec_helper', __FILE__)
+
+require 'json'
+
+describe "String#to_f" do
+  
+  it "resists CVE-2013-4164 by converting very long Strings to a Float" do
+    "1.#{'1'*1000000}".to_f.should be_close(1.1111111111111112, TOLERANCE)
```

Comment:
Is this enough to test for this issue? Is CVE-2013-4164 a bug in float parsing?
I know a variant where essentially Float parsing time is quadratic to the input length, potential for DDoS.